fdbrpc/tests/fdbrpc_bench.actor.cpp

```diff
@@ -0,0 +1,232 @@
+};
+
+ACTOR Future<Void> echoServer() {
+	state EchoServerInterface echoServer;
+	state StatCounter c;
+	echoServer.getInterface.makeWellKnownEndpoint(WLTOKEN_ECHO_SERVER, TaskPriority::DefaultEndpoint);
+	state Future<Void> next = delay(10);
+	loop {
+		try {
+			choose {
+				when(GetInterfaceRequest req = waitNext(echoServer.getInterface.getFuture())) {
+					req.reply.send(echoServer);
+				}
+				when(EchoRequest req = waitNext(echoServer.echo.getFuture())) {
+					req.reply.send(req.message);
+					c.inc();
+				}
+				when(wait(next)) {
+					next = delay(10);
+					std::cout << "Throughput: " << c.avg() << " req/sec" << std::endl;
+				}
+			}
+		} catch (Error& e) {
+			if (e.code() != error_code_operation_obsolete) {
+				fprintf(stderr, "Error: %s\n", e.what());
+				throw e;
+			}
+		}
+	}
+}
+
+const int payload_size_bytes = 1024 * 1000;
+
+std::string randString(int size) {
+	const std::string charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+	const int charsetLength = charset.length();
+	std::string result;
+
+	// Seed the random number generator
+	std::srand(static_cast<unsigned int>(std::time(nullptr)));
+
+	for (int i = 0; i < size; ++i) {
+		result += charset[std::rand() % charsetLength];
+	}
+
+	return result;
+}
+
+ACTOR Future<Void> echoClient() {
+	state EchoServerInterface server;
+	server.getInterface =
+	    RequestStream<GetInterfaceRequest>(Endpoint::wellKnown({ serverAddress }, WLTOKEN_ECHO_SERVER));
+	EchoServerInterface s = wait(server.getInterface.getReply(GetInterfaceRequest()));
+	server = s;
+	state std::string payload = randString(payload_size_bytes);
+
+	while (true) {
+		state int duration_seconds = 10;
+		state int request_count = 0;
+
+		state std::chrono::time_point<std::chrono::steady_clock> start_time = std::chrono::steady_clock::now();
+		state std::chrono::time_point<std::chrono::steady_clock> end_time =
+		    start_time + std::chrono::seconds(duration_seconds);
+
+		while (std::chrono::steady_clock::now() < end_time) {
+			EchoRequest echoRequest;
+			echoRequest.message = payload;
+			std::string echoMessage = wait(server.echo.getReply(echoRequest));
+			++request_count;
+		}
+		std::cout << "Sent " << request_count << " requests in " << request_count / duration_seconds << " /second"
+		          << std::endl;
+	}
+}
+
+std::unordered_map<std::string, std::function<Future<Void>()>> actors = {
+	{ "echoServer", &echoServer },
+	{ "echoClient", &echoClient },
+};
+} // namespace fdbrpc_bench
+
+int main(int argc, char* argv[]) {
+	using namespace fdbrpc_bench;
+
+	bool isServer = false;
+	std::string port;
+	std::vector<std::function<Future<Void>()>> toRun;
+
+	// parse arguments
+	for (int i = 1; i < argc; ++i) {
+		std::string arg(argv[i]);
+		if (arg == "echoServer")
+			isServer = true;
+		auto actor = actors.find(arg);
+		if (actor == actors.end()) {
+			std::cout << format("Error: actor %s does not exist\n", arg.c_str());
+			return 1;
```

Comment:
can you print the usage as well?

Reply:
boost program options does that with new changes.